txcache/selectionTracker.go

```diff
@@ -0,0 +1,147 @@
+package txcache
+
+import (
+	"bytes"
+	"math/big"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+)
+
+type accountBreadcrumb struct {
+	initialNonce    uint64
+	lastNonce       uint64
+	consumedBalance *big.Int
+}
+
+type trackedBlock struct {
```

Comment:
Maybe have this in a separate file, along with the constructor `newTrackedBlock`?